apps/crn-server/scripts/export-entity.ts

```diff
@@ -86,10 +86,29 @@ const getController = (entity: keyof EntityResponsesCRN) => {
 const transformRecords = <T extends EntityResponsesCRN, K extends keyof T>(
```

Comment:
I tried to: 
1. Get rid of casting
2. Remove unnecessary ts-ignore

I've achieved it by doing this:
```typescript
const transformRecords = <T extends EntityResponsesCRN, K extends keyof T>(
  record: T[K] extends EntityResponsesCRN[keyof EntityResponsesCRN]
    ? T[K] & { id: string }
    : never,
  type: K extends keyof EntityResponsesCRN ? K : never,
) => {
  const payload = {
    ...record,
    objectID: record.id,
    __meta: {
      type,
    },
  };

  if (type === 'research-output' && 'subtype' in record) {
    const subtype = record.subtype;

    return {
      ...payload,
      _tags: [
        ...record.methods,
        ...record.organisms,
        ...record.environments,
        ...(subtype ? [subtype] : []),
        ...record.keywords,
      ],
    };
  }
```

What's changed:
* added conditional type checks for function arguments to narrow the types down
* removed ts-ignore comment and replace mutation with a simple return
* added extra condition to recognise the type of `record`, then removed all casts